router/core/graphql_prehandler.go

```diff
@@ -651,19 +628,26 @@ func (h *PreHandler) parseRequestOptions(r *http.Request, clientInfo *ClientInfo
 }
 
 func (h *PreHandler) internalParseRequestOptions(r *http.Request, clientInfo *ClientInfo, requestLogger *zap.Logger) (resolve.ExecutionOptions, resolve.TraceOptions, error) {
-	if h.developmentMode {
-		return h.parseRequestExecutionOptions(r), h.parseRequestTraceOptions(r), nil
-	}
-	if h.enableRequestTracing && clientInfo.WGRequestToken != "" && h.routerPublicKey != nil {
-		_, err := jwt.Parse(clientInfo.WGRequestToken, func(token *jwt.Token) (interface{}, error) {
-			return h.routerPublicKey, nil
-		}, jwt.WithValidMethods([]string{jwt.SigningMethodES256.Name}))
-		if err != nil {
-			requestLogger.Error(fmt.Sprintf("failed to parse request token: %s", err.Error()))
-			return resolve.ExecutionOptions{}, resolve.TraceOptions{}, err
+	// Determine if we should enable request tracing / query plans at all
+	if h.enableRequestTracing {
+		// In dev mode we always enable tracing / query plans
+		if h.developmentMode {
+			return h.parseRequestExecutionOptions(r), h.parseRequestTraceOptions(r), nil
+		}
+		// If the client has a valid request token and we have a public key from the controlplane
+		if clientInfo.WGRequestToken != "" && h.routerPublicKey != nil {
```

Comment:
do we want to move this logic somewhere else and encapsulate it better?

Reply:
I think it is fine for now.